src/lib/arch/aarch64/Instruction_execute.cc

```diff
@@ -4830,6 +4896,16 @@ void Instruction::execute() {
       results[0] = mulhi(x, y);
       break;
     }
+    case Opcode::AArch64_USHLLv16i8_shift: {  // ushll2 vd.8h, vn.16b, #imm
+      const uint8_t* n = operands[0].getAsVector<uint8_t>();
+      int64_t shift = metadata.operands[2].imm;
+      uint16_t out[8] = {0};
+      for (int i = 0; i < 8; i++) {
+        out[i] = (n[i + 8] << shift) & 0xff;
```

Comment:
Is there any need for the `& 0xff`? As `shift` max value is source element width in bits (i.e. here it will be 8), which is always half of the destination element's width (here this is 16). Could possibly do a `static_cast<uint16_t>` to make it clear what the output format is?

Additionally, doing this operation with `0xff` would confine it to 8 bit values? i.e. `0b1111111111111111 & 0xff = 0b0000000011111111`

Reply:
I tested on tx2 and you're right, the correct behaviour of these instructions does not restrict the width of shifted values to the width of source.
I did not know earlier implementation of `ushll vd.4s, vn.4h, #imm` was wrong, so I used it as a reference.
I fixed both my implementation and early implementation.